service/src/test/java/uk/nhs/adaptors/gpc/consumer/FindAPatientDocsTest.java

```diff
@@ -0,0 +1,76 @@
+package uk.nhs.adaptors.gpc.consumer;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.util.zip.GZIPOutputStream;
+
+import org.junit.jupiter.api.Test;
+
+import lombok.SneakyThrows;
+import uk.nhs.adaptors.gpc.consumer.filters.FindAPatientDocsGatewayFilterFactory;
+import uk.nhs.adaptors.gpc.consumer.utils.FindAPatientDocsUtil;
+
+public class FindAPatientDocsTest {
+    private static final String GPC_BASE_URL = "http://localhost:8110";
+    private static final String GPC_CONSUMER_BASE_URL = "http://localhost:8080";
+    private static final String RESPONSE_BODY_TEMPLATE = "{\"url\": \"%s/example\"}";
+    private static final String GPC_RESPONSE_BODY = String.format(RESPONSE_BODY_TEMPLATE, GPC_BASE_URL);
+
+    @Test
+    public void When_GPCBaseUrlPresentInBody_Expect_GPCConsumerBaseUrlReplacementInBody() {
+        var expectedGpcConsumerResponse = String.format(RESPONSE_BODY_TEMPLATE, GPC_CONSUMER_BASE_URL);
+        FindAPatientDocsGatewayFilterFactory.Config config = new FindAPatientDocsGatewayFilterFactory.Config();
+        config.setGpcUrl(GPC_BASE_URL);
+        config.setGpcConsumerUrl(GPC_CONSUMER_BASE_URL);
+
+        var replacedUrlBody = FindAPatientDocsUtil.replaceUrl(config, GPC_RESPONSE_BODY);
+        assertThat(replacedUrlBody).isEqualTo(expectedGpcConsumerResponse);
+    }
+
+    @Test
+    public void When_ZipPramsString_Expect_ByteArrayOutputStream() throws Exception {
+        var os = FindAPatientDocsUtil.zipStringToOutputStream(GPC_RESPONSE_BODY);
```

Comment:
Could the variable names 'os' and 'is' be changed to more obvious for easier reading if that's ok
Seen these names used across this class

Reply:
os and is are pretty standard for outputstream and inputstream